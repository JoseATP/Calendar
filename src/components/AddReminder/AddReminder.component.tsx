import { FC } from "react";
import Button, {
  BUTTON_TYPE_CLASSES,
} from "../../components/Button/Button.component";
import { AddReminderContainer, DateLabel } from "./AddReminder.styles";
import { DATE_OPTIONS } from "../../constants/date.constants";
import { useNavigate } from "react-router-dom";

type AddReminderProps = {
  selectedDate: Date;
};

const AddReminder: FC<AddReminderProps> = ({ selectedDate }) => {
  const navigate = useNavigate();

  return (
    <AddReminderContainer>
      <DateLabel>
        {new Date(selectedDate).toLocaleString("en-MX", DATE_OPTIONS)}
      </DateLabel>
      <Button
        buttonType={BUTTON_TYPE_CLASSES.gradientButton}
        onClick={() => navigate("addReminder")}
      >
        Add reminder
      </Button>
    </AddReminderContainer>
  );
};

export default AddReminder;
