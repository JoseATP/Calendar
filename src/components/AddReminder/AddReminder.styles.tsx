import styled from "styled-components";

export const AddReminderContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-around;
  margin: 20px 0px;
`;

export const DateLabel = styled.h2``;
