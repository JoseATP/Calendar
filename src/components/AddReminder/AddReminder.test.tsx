import { render, fireEvent, screen } from "@testing-library/react";
import AddReminder from "./AddReminder.component";
import { useNavigate } from "react-router-dom";

const mockedUsedNavigate = jest.fn();
jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockedUsedNavigate,
}));

describe("AddReminder", () => {
  it("Should render AddReminder component correctly", () => {
    const { asFragment } = render(<AddReminder selectedDate={new Date()} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("Should navigate to addReminder router correctly", () => {
    render(<AddReminder selectedDate={new Date()} />);

    const button = screen.getByText("Add reminder");

    fireEvent.click(button);

    expect(useNavigate()).toHaveBeenCalledTimes(1);
    expect(useNavigate()).toHaveBeenCalledWith("addReminder");
  });
});
