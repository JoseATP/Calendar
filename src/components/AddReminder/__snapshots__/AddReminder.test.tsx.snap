// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddReminder Should render AddReminder component correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-eDLJxc YCoke"
  >
    <h2
      class="sc-jTQDnj"
    >
      Tuesday, March 7, 2023
    </h2>
    <button
      class="sc-beySbM sc-hLQRIN haKdpM fTsCQI"
    >
      Add reminder
    </button>
  </div>
</DocumentFragment>
`;
